src/Common/tests/gtest_generate_random_by_regexp.cpp

```diff
@@ -0,0 +1,101 @@
+#include <Common/MatchGenerator.h>
+#include <Common/ObjectStorageKeyGenerator.h>
+#include <Common/Stopwatch.h>
+#include <Common/Exception.h>
+
+#include <gtest/gtest.h>
+
+void routine(String s)
+{
+    std::cerr << "case '"<< s << "'";
+    auto gen = DB::RandomStringGeneratorByRegexp(s, /*logging*/ true);
```

Comment:
does it make sense to test if the generated string passes regex match using the same regex?

Reply:
I think it does. 
At least it it the cheapest way to check that result string really matches the requested pattern.
For example, generation is broken and generates empty string. Or it generates some prefix and stopes. re2:match would fail on such wrong results.